### ISSUE TITLE
Infinite loop detected when run rubocop -a to some code with miss alignment 'end' 

### ISSUE THREAD

**Issue by t-ashula** [rubocop-hq/rubocop#6202]
--------

## Expected behavior

correct successfully 

## Actual behavior

infinite loop detected in source code

## Steps to reproduce the problem

1. create test.rb with this code

```ruby
def method1
  'foo'
end

def method2
  'bar'
rescue
  'baz' end
```

2.  ` rubocop -a test.rb`

## RuboCop version

```
$rubocop -V
0.58.2 (using Parser 2.5.1.2, running on ruby 2.5.1 x86_64-linux)
```

**Comment by rrosenblum** [comment]
The cops that run auto-correction on this are `Layout/RescueEnsureAlignment`, `Style/RescueStandardError`, and `Layout/IndentationWidth`. It seems like `Layout/RescueEnsureAlignment` is the primary culprit. The cop appears to be trying to align the `rescue` based on the position of `end`.

```ruby
test.rb:9:1: C: [Corrected] Layout/RescueEnsureAlignment: rescue at 9, 0 is not aligned with end at 10, 8.
rescue
^^^^^^
test.rb:9:1: C: [Corrected] Style/RescueStandardError: Avoid rescuing without specifying an error class.
rescue
^^^^^^
test.rb:9:9: C: [Corrected] Layout/RescueEnsureAlignment: rescue at 9, 8 is not aligned with end at 10, 16.
        rescue StandardError
        ^^^^^^
test.rb:9:17: C: [Corrected] Layout/RescueEnsureAlignment: rescue at 9, 16 is not aligned with end at 10, 24.
                rescue StandardError
                ^^^^^^
test.rb:9:25: C: [Corrected] Layout/RescueEnsureAlignment: rescue at 9, 24 is not aligned with end at 10, 32.
                        rescue StandardError
                        ^^^^^^
test.rb:9:33: C: [Corrected] Layout/RescueEnsureAlignment: rescue at 9, 32 is not aligned with end at 10, 40.
                                rescue StandardError
                                ^^^^^^
```

**Comment by rrosenblum** [comment]
I'm looking into this issue. It may take me a little while to fix it.
